app/scripts/controllers/preferences-controller.ts

```diff
@@ -149,125 +155,294 @@ export type PreferencesControllerState = {
   textDirection?: string;
 };
 
-export default class PreferencesController {
-  store: ObservableStore<PreferencesControllerState>;
+/**
+ * Function to get default state of the {@link PreferencesController}.
+ */
+export const getDefaultPreferencesControllerState =
+  (): PreferencesControllerState => ({
+    selectedAddress: '',
+    useBlockie: false,
+    useNonceField: false,
+    usePhishDetect: true,
+    dismissSeedBackUpReminder: false,
+    useMultiAccountBalanceChecker: true,
+    useSafeChainsListValidation: true,
+    // set to true means the dynamic list from the API is being used
+    // set to false will be using the static list from contract-metadata
+    useTokenDetection: true,
+    useNftDetection: true,
+    use4ByteResolution: true,
+    useCurrencyRateCheck: true,
+    useRequestQueue: true,
+    openSeaEnabled: true,
+    securityAlertsEnabled: true,
+    watchEthereumAccountEnabled: false,
+    bitcoinSupportEnabled: false,
+    bitcoinTestnetSupportEnabled: false,
+    ///: BEGIN:ONLY_INCLUDE_IF(keyring-snaps)
+    addSnapAccountEnabled: false,
+    ///: END:ONLY_INCLUDE_IF
+    advancedGasFee: {},
+
+    // WARNING: Do not use feature flags for security-sensitive things.
+    // Feature flag toggling is available in the global namespace
+    // for convenient testing of pre-release features, and should never
+    // perform sensitive operations.
```

Comment:
do you think it would make sense to move these warnings and comments to the `PreferencesControllerState` type declaration?